drivers/nrf24l01p/gnrc_netif_nrf24l01p.c

```diff
@@ -0,0 +1,221 @@
+ *            transceiver
+ *
+ * @author Hauke Petersen <hauke.petersen@fu-berlin.de>
+ * @author Peter Kietzmann <peter.kietzmann@haw-hamburg.de>
+ * @author Fabian Hüßler <fabian.huessler@ovgu.de>
+ * @}
+ */
+#include <assert.h>
+
+#define ENABLE_DEBUG    (0)
+#include "debug.h"
+
+#include "net/gnrc.h"
+#include "nrf24l01p_internal.h"
+
+/**
+ * @brief   Broadcast/Multicast flag
+ */
+#define BCAST  (GNRC_NETIF_HDR_FLAGS_BROADCAST | GNRC_NETIF_HDR_FLAGS_MULTICAST)
+
+static gnrc_pktsnip_t *_nrf24l01p_pkt_recv(gnrc_netif_t *netif)
+{
+    /* get upper size limit */
+    int upper_frame_len = netif->dev->driver->recv(netif->dev, NULL, 0, NULL);
+
+    uint8_t frame_buffer[upper_frame_len];
+    /* copy the payload into the packet buffer */
+    int frame_len = netif->dev->driver->recv(netif->dev, frame_buffer,
+                                                 upper_frame_len, NULL);
+    if(frame_len <= 0) {
+        return NULL;
+    }
+
+    /* allocate space for the packet in the pktbuf */
+    gnrc_pktsnip_t *frame = gnrc_pktbuf_add(NULL, NULL, frame_len,
+                                            NRF24L01P_UPPER_LAYER_PROTOCOL);
+    if (!frame) {
+        DEBUG("[nrf24l01p] recv: unable to allocate space in the pktbuf\n");
+        return NULL;
+    }
+    memcpy(frame->data, frame_buffer, frame_len);
+    return frame;
+}
+
+/**
+ * @brief   Receives a @ref net_gnrc_pkt "packet" from the network interface
+ *
+ * @pre `netif != NULL`
+ *
+ * @note The function takes the bytes received via netdev_driver_t::recv()
+ *       from gnrc_netif_t::dev and re-formats it to a
+ *       @ref net_gnrc_pkt "packet" containing a @ref net_gnrc_netif_hdr
+ *       and a payload header in receive order.
+ *
+ * @param[in] netif The network interface.
+ *
+ * @return  The packet received. Contains the payload (with the type marked
+ *          accordingly) and a @ref net_gnrc_netif_hdr in receive order.
+ * @return  NULL, if @ref net_gnrc_pktbuf was full.
+ */
+static gnrc_pktsnip_t *nrf24l01p_adpt_recv(gnrc_netif_t *netif)
+{
+    netdev_t *dev = netif->dev;
+    uint8_t header_len = 0;
+    gnrc_pktsnip_t *frame;
+    gnrc_pktsnip_t *hdr;
+    gnrc_netif_hdr_t *netif_hdr;
+    shockburst_hdr_t sb_hdr;
+
+    assert(dev);
+
+    if (!(frame = _nrf24l01p_pkt_recv(netif))) {
+        return NULL;
+    }
+    uint8_t bcast_addr[] = NRF24L01P_BROADCAST_ADDR;
+    sb_hdr.addr_width = ((uint8_t *)(frame->data))[header_len++];
+    uint8_t dst_addr_width = sb_hdr_get_dst_addr_width(&sb_hdr);
+    memcpy(sb_hdr.dst_addr,
+           ((uint8_t *)(frame->data)) + header_len,
+           dst_addr_width);
+    header_len += dst_addr_width;
+#ifdef NRF24L01P_CUSTOM_HEADER
+    uint8_t src_addr_width = sb_hdr_get_src_addr_width(&sb_hdr);
+    memcpy(sb_hdr.src_addr,
+           ((uint8_t *)(frame->data)) + header_len,
+           src_addr_width);
+    header_len += src_addr_width;
+#endif
```

Comment:
Can you rebase against current master and use the following instead:

```C
    uint8_t src_addr_width = 0;
    if (IS_USED(NRF24L01P_CUSTOM_HEADER)) {
        uint8_t src_addr_width = sb_hdr_get_src_addr_width(&sb_hdr);
        memcpy(sb_hdr.src_addr,
               ((uint8_t *)(frame->data)) + header_len,
               src_addr_width);
        header_len += src_addr_width;
    }
```

Reply:
Btw: Is there indeed a need for not using the L2 header that contains the source address? I'd personally just always add the source address until someone explicitly asks for allowing to send the frames without. This would make the code a bit more readable; and hopefully nobody ever complains :grin:

Reply:
Including the source address is not the default behavior  for ShockBurst or Enhanced ShockBurst.
It takes at most 6 bytes of the 32 byte FIFO. If someone uses the `nrf24l01p` with another driver (not RIOT), the transceiver would not know what these ~6 bytes are, causing incompatibility.
I would not do that.

Ok, I will rebase and use your suggested change.

Reply:
> Can you rebase against current master and use the following instead:
> 
> ```c
>     uint8_t src_addr_width = 0;
>     if (IS_USED(NRF24L01P_CUSTOM_HEADER)) {
>         uint8_t src_addr_width = sb_hdr_get_src_addr_width(&sb_hdr);
>         memcpy(sb_hdr.src_addr,
>                ((uint8_t *)(frame->data)) + header_len,
>                src_addr_width);
>         header_len += src_addr_width;
>     }
> ```

I have a problem with mixing
```C
#if IS_USED(NRF24L01P_CUSTOM_HEADER)
    uint8_t src_addr[NRF24L01P_MAX_ADDR_WIDTH];
#endif
```
in `shockburst_hdr_t` and
```C
if (IS_USED(NRF24L01P_CUSTOM_HEADER)) {
...
}
```
because the member `src_addr` does not exist if `NRF24L01P_CUSTOM_HEADER`
is not defined, but it is used in code.
I´d rather stick to consistent usage of `#if IS_USED(NRF24L01P_CUSTOM_HEADER)`, if don´t mind